### PR TITLE
New design: pagination.

### PR DESCRIPTION
**Pull request by isoos**

### CODE REVIEW

File: app/lib/frontend/templates.dart [hunk 1]
```diff
@@ -567,26 +575,24 @@ abstract class PageLinks {
 
     final bool hasPrevious = currentPage > 1;
     results.add({
-      'state': {'state': hasPrevious ? null : 'disabled'},
```
Comment (kevmoo) [inline app/lib/frontend/templates.dart]: Why the churn here?

File: app/test/frontend/golden/flutter_packages_index_page2.html [hunk 2]
```diff
@@ -114,20 +114,22 @@
 
 <div class="pagination">
   <ul>
-      <li class="">
-        <a href="/packages?page=1">
+      <li class=" ">
+        <a href="&#47;packages?page=1">
```
Comment (kevmoo) [inline app/test/frontend/golden/flutter_packages_index_page2.html]: This seems broken...

File: app/test/frontend/golden/flutter_packages_index_page2.html [hunk 3]
```diff
@@ -114,20 +114,22 @@
 
 <div class="pagination">
   <ul>
-      <li class="">
-        <a href="/packages?page=1">
+      <li class=" ">
```
Comment (kevmoo) [inline app/test/frontend/golden/flutter_packages_index_page2.html]: Why the extra whitespace?

File: app/test/frontend/golden/flutter_packages_index_page2.html (hunk 2 above)
Comment (isoos) [reply app/test/frontend/golden/flutter_packages_index_page2.html]: We have `&` in these URLs, and that needs to be escaped. Unfortunately the HtmlEscaper is a bit too eager to do other escaping, but it is valid XML attribute, and the browser treats it OK.

File: app/lib/frontend/templates.dart (hunk 1 above)
Comment (isoos) [reply app/lib/frontend/templates.dart]: I'm switching to simpler template preparation and mustache structure (esp. for the new pagination).

File: app/test/frontend/golden/flutter_packages_index_page2.html (hunk 3 above)
Comment (isoos) [reply app/test/frontend/golden/flutter_packages_index_page2.html]: Simpler rendering has its downsizes. Do you prefer the tradeoff for the better HTML output?

File: app/test/frontend/golden/flutter_packages_index_page2.html (hunk 2 above)
Comment (isoos) [reply app/test/frontend/golden/flutter_packages_index_page2.html]: Switched to an attribute-specific HTML escaper, this is now better.

File: app/lib/frontend/templates.dart [hunk 4]
```diff
@@ -26,6 +26,7 @@ String _escapeAngleBrackets(String msg) =>
     const HtmlEscape(HtmlEscapeMode.ELEMENT).convert(msg);
 
 const HtmlEscape _HtmlEscaper = const HtmlEscape();
+const HtmlEscape _attrEscaper = const HtmlEscape(HtmlEscapeMode.ATTRIBUTE);
```
Comment (mkustermann) [inline app/lib/frontend/templates.dart]: Could be consistent with `_Html` vs. `_attr`. I know our new style guide is very confusing regarding constants, previously we had nice `CONST_VALUE` names.

File: app/lib/frontend/templates.dart [hunk 5]
```diff
@@ -404,6 +405,14 @@ class TemplateService {
     return _renderTemplate('layout', values, escapeValues: false);
   }
 
+  /// Renders the `views/pagination_new.mustache` template.
+  String renderNewPagination(PageLinks pageLinks) {
+    final values = {
+      'page_links': pageLinks.hrefPatterns(),
+    };
+    return _renderTemplate('pagination_new', values, escapeValues: false);
```
Comment (mkustermann) [inline app/lib/frontend/templates.dart]: Maybe have a `v2/pagination` here.

File: app/lib/frontend/templates.dart (hunk 4 above)
Comment (isoos) [reply app/lib/frontend/templates.dart]: I'll do a follow-up PR to rename `_HtmlEscaper` -> `_htmlEscaper`.

**Pull request merged**